fuzz.go

```diff
@@ -221,6 +227,9 @@ func (f *Fuzzer) Fuzz(obj interface{}) {
 // obj must be a pointer. Only exported (public) fields can be set (thanks, golang :/ )
 // Intended for tests, so will panic on bad input or unimplemented fields.
 func (f *Fuzzer) FuzzNoCustom(obj interface{}) {
+	f.fuzzLock.Lock()
```

Comment:
I think this will break reentrant callers?

Reply:
recursion is fine and encouraged, multi threaded use is not allowed though.

Reply:
Actually what I said doesn't match the one comment about thread safety in this file.

Reply:
I am confused, how would an external user make this recurse? The multithreaded use is something I only found because it paniced with an out of bounds in the prng. The reason I want to keep reusing a fuzzer is that the prng doesn't yield a lot of entropy when always initialized with the timestamp in a bunch of tests that run in a short timeframe

Reply:
https://github.com/google/gofuzz/blob/master/example_test.go#L195

Did you run all the tests? I'd expect a number of them to deadlock.

Reply:
Maybe an OK solution is to have a `func (f *Fuzzer) NextSeed() *Fuzzer` which makes a new random source and gives a new fuzzer with it which is safe to call concurrently with the parent.

Reply:
> Did you run all the tests? I'd expect a number of them to deadlock.

Yeah, they pass fine:
```
$ go test -race ./... -count=1
ok  	github.com/google/gofuzz	0.060s
ok  	github.com/google/gofuzz/bytesource	0.021s
```

Reply:
Oh, I see, the continue has its own entrypoints.

Reply:
Someone failing to correctly use the Continue in a custom fuzz function could still cause a deadlock. That's improper use but I don't really want to break any consumers of this library. hm.

Reply:
@lavalamp are you ok with interfacing out the underlying rand.Rand and doing the locking there? That shouldn't break anyone

Reply:
I think I prefer your current solution because it shouldn't have a performance penalty. I'm super busy, give me some more time to think.